src/main/java/net/schmizz/sshj/userauth/keyprovider/PKCS5KeyFile.java

```diff
@@ -126,85 +146,56 @@ protected KeyPair readKeyPair()
                 if (line.startsWith("-----BEGIN ") && line.endsWith(" PRIVATE KEY-----")) {
                     int end = line.length() - 17;
                     if (end > 11) {
-                        type = line.substring(11, line.length() - 17);
+                        String s = line.substring(11, line.length() - 17);
+                        if ("RSA".equals(s)) {
+                            type = KeyType.RSA;
+                        } else if ("DSA".equals(s)) {
+                            type = KeyType.DSA;
+                        } else if ("DSS".equals(s)) {
+                            type = KeyType.DSA;
+                        } else {
+                            throw new FormatException("Unrecognized PKCS5 key type: " + s);
+                        }
                     } else {
-                        type = "UNKNOWN";
+                        throw new FormatException("Bad header; possibly PKCS8 format?");
                     }
                 } else if (line.startsWith("-----END")) {
                     break;
                 } else if (type != null) {
                     if (line.startsWith("Proc-Type: ")) {
                         if (!"4,ENCRYPTED".equals(line.substring(11))) {
-                            throw new IOException("Unrecognized Proc-Type: " + line.substring(11));
+                            throw new FormatException("Unrecognized Proc-Type: " + line.substring(11));
                         }
                     } else if (line.startsWith("DEK-Info: ")) {
                         int ptr = line.indexOf(",");
                         if (ptr == -1) {
-                            throw new IOException("Unrecognized DEK-Info: " + line.substring(10));
+                            throw new FormatException("Unrecognized DEK-Info: " + line.substring(10));
                         } else {
                             String algorithm = line.substring(10,ptr);
```

Comment:
That one seems like a false positive, i.e. this is a local variable not a field.